source/dotnet/Library/AdaptiveCards.Rendering.Wpf/Helpers/ImageExtensions.cs

```diff
@@ -86,17 +87,57 @@ public object ConvertBack(object value, Type targetType, object parameter, Cultu
             }
         }
 
-        public static async void SetBackgroundSource(this Grid grid, Uri url, AdaptiveRenderContext context)
+        public static Uri ResolveFinalAbsoluteUri(Uri uri, String imageBaseUrl)
         {
```

Comment:
Can't this function and the one in the html renderer be consolidated in to the same one in the shared code?

Reply:
yeah this change actually makes so much sense